### PR TITLE
Create a product doesn't imply a save in database anymore

### PR DESCRIPTION
**Pull request by Quetzacoalt91**
:warning: This pull-request is a POC and not all tests have been executed. You can review it, test it, but do not merge it !

| Questions | Answers |
| --- | --- |
| Branch? | develop |
| Description? | Improvement |
| Type? | Improvement |
| Category? | BO |
| BC breaks? | Nope |
| Deprecations? | Nope |
| Fixed ticket? | None |
| How to test? | Click on the button "Add a new product" from the catalog. Nothing should be inserted in database until we add a title. |

### CODE REVIEW

File: src/PrestaShopBundle/Resources/config/admin/routing_product.yml [hunk 1]
```diff
@@ -58,8 +58,7 @@ admin_product_form:
         _legacy_controller: AdminProducts
         _legacy_param_mapper_class: PrestaShop\PrestaShop\Adapter\Product\AdminProductDataProvider
         _legacy_param_mapper_method: mapLegacyParametersProductForm
-    requirements:
-        id: \d+
+        id: 0
```
Comment (mickaelandrieu) [inline src/PrestaShopBundle/Resources/config/admin/routing_product.yml]: the only available value for this route is `0`: is that what you want ?

Comment (Quetzacoalt91) [inline src/PrestaShopBundle/Resources/config/admin/routing_product.yml]: I removed the "requirements" fields, it should become a default value from now.

**Comment by Quetzacoalt91** [comment]
This PR has been updated.

It now includes:
* A fix to display the initial quantity to 0
* Disable the automatic processing of the images until the product has been saved
* The update of the browser history to add the product id in the URL with the JS (without redirecting the merchant !)

File: admin-dev/themes/default/js/bundle/product/form.js [hunk 2]
```diff
@@ -720,6 +729,12 @@ var form = (function() {
         $('*.has-danger').removeClass('has-danger');
       },
       success: function(response) {
+        var id_product = $('#form_id_product').val();
+        if (!id_product) {
+            $('#form_id_product').val(response.product.id);
+            $('#form_id_product').change();
+            window.history.pushState({"pageTitle":document.title},"", "/PrestaShop-dev/admin-dev/product/form/"+response.product.id);
```
Comment (Quetzacoalt91) [inline admin-dev/themes/default/js/bundle/product/form.js]: :warning: To be removed

**Comment by Quetzacoalt91** [comment]
This PR now includes:
* A workaround for dropzone.js to upload every image in queue, not only the 2 first
* Default values for category and visibility of the product
* A test to avoid JS errors on empty response from the controller

**Comment by vincentbz** [comment]
So, when I create a product :+1:  
 - the URL doesn't have an ID, the product is not inserted into the DB
 - the name field is empty and that's pretty cool
 - If I save for the first time, it adds an ID to the product and inserts it in DB



If I delete or duplicate a product from the product page, before I reload the page : 
image

If I reload the page before these actions, it's OK I don't have the error

**Comment by vincentbz** [comment]
A very good thing that created a small bug : 
Now if have two languages, FR and EN, and I enter the name in En, it will display it in FR if I don't enter a specific name in FR.

Buuut, if I enter a name in FR, and erase it, it still keep that FR name.
image
image

Thanks :)

**Comment by vincentbz** [comment]
Cool thing to add : 
At product creation, the friendly URL is empty. As the product is officialy created when we save for the first time, is it possible to create the friendly with the first save too ? (but only at first save, like the ID that appears in the URL)

Thanks :)

**Pull request closed without merging**